apps/dashboard/src/components/workflow-editor/step-utils.ts

```diff
@@ -18,28 +8,39 @@ import {
   DEFAULT_CONTROL_DIGEST_UNIT,
   STEP_TYPE_LABELS,
 } from '@/utils/constants';
-
-export const getFirstBodyErrorMessage = (issues?: StepIssuesDto) => {
-  const stepIssuesArray = Object.entries({ ...issues?.body });
-  if (stepIssuesArray.length > 0) {
-    const firstIssue = stepIssuesArray[0];
-    const errorMessage = firstIssue[1]?.message;
-    return errorMessage;
-  }
-};
+import type {
+  StepContentIssue,
+  StepCreateDto,
+  StepIssuesDto,
+  StepUpdateDto,
+  UpdateWorkflowDto,
+  WorkflowResponseDto,
+} from '@novu/shared';
+import { StepTypeEnum } from '@novu/shared';
+import { flatten } from 'flat';
 
 export const getFirstControlsErrorMessage = (issues?: StepIssuesDto) => {
   const controlsIssuesArray = Object.entries({ ...issues?.controls });
   if (controlsIssuesArray.length > 0) {
     const firstIssue = controlsIssuesArray[0];
     const contentIssues = firstIssue?.[1];
-    const errorMessage = contentIssues?.[0]?.message;
-    return errorMessage;
+    const error = contentIssues?.[0];
+    return error;
+  }
+};
+
+export const getFirstIntegrationErrorMessage = (issues?: StepIssuesDto) => {
```

Comment:
The step issue interface should be the same under controls or integrations key, so that we can reuse the utility instead of duplicating it.

Reply:
Does this work for u?
```
getFirstErrorMessage(step.issues, 'controls')
```